jepsen/src/jepsen/core.clj

```diff
@@ -257,7 +308,7 @@
   nemesis completion, and tears down nemesis."
   [test & body]
   ; Initialize nemesis
-  `(let [nemesis# (client/setup! (:nemesis ~test) ~test nil)]
+  `(let [nemesis# (client/open-compat! (:nemesis ~test) ~test nil)]
```

Comment:
Shoot, I didn't even think about this. Nemeses aren't bound to any particular node, so `open!` doesn't make... logical sense for them. In fact, we call setup! with node *nil*. Do you think we should make nemeses their own protocol, since they don't really follow the same lifecycle as clients anyway?

Reply:
Giving nemeses their own protocol makes a lot of sense to me. They get their own worker setup before all of these codepaths, so sharing the protocol here makes them intertwine in odd ways that were a constant edge case while implementing.

I can scope this out and ship it before we land 1.7?

Reply:
Yeah, I think you're right. Let's do a Nemesis protocol, but fall back to calling client/setup! and client/teardown! when nemesis funs aren't available on a nemesis, for compatibility. Same story as the new Client functions.